src/twisted/conch/test/test_conch.py

```diff
@@ -294,6 +307,9 @@ class ConchServerSetupMixin:
     if not cryptography:
         skip = "can't run without cryptography"
 
+    if not hasDsa:
+        skip = "needs ssh supporting dsa"
```

Comment:
As above, IMO this skips too much.  Most of these tests still make conceptual sense to run with an OpenSSH that doesn't support DSA - we just need to run them in a way that doesn't rely on DSA.  I haven't checked in detail, but I'd have thought most of them could just use an RSA key instead.

Reply:
I moved the `if hasDsa` check to `OpenSSHClientMixin` class. It skips two additional tests except of those from original report, `twisted.conch.test.test_conch.OpenSSHClientRekeyTests.test_clientRekey` and `twisted.conch.test.test_conch.OpenSSHKeyExchangeTests.test_unsupported_algorithm`. I think it is acceptable sacrifice :)